### ISSUE TITLE
Deploying Index.html succeeds however isn't served due to case sensitive file system

### ISSUE THREAD

**Issue by tomaustin700** [Azure/static-web-apps#624]
If a site is deployed and the index page is Index.html instead of index.html everything looks to have deployed correctly however Azure will just return 404 when navigating to the static web app. After discussing with @anthonychu on Twitter this is due to file system being case sensitive and the check for index.html is not case sensitive.

**To Reproduce**
1. Have a repo with an Index.html instead of index.html
2. Deploy the site using GitHub Actions
3. Attempt to view your deployed site and Azure just returns 404

**Expected behavior**
GitHub action should error and block deployment but give a meaningful error message

**Additional context**
See https://twitter.com/tomaustin700/status/1457373532070031363

**Comment by andersme** [comment]
We are also running into this issue. Some of our CSS files were referenced using the wrong casing and failed to load. This really should be address since URLs should be case insensitive.

**Comment by miwebst** [comment]
URL paths are supposed to be case sensitive. It matters for things like SEO scoring, routing, etc.

**Comment by andersme** [comment]
> URL paths are supposed to be case sensitive. It matters for things like SEO scoring, routing, etc.

Yes, it is more of my opinion that they should be case insensitive. However, in this case, maybe there needs to be an option in Azure to say whether you want to use case-insensitive matching or not.

**Comment by onionhammer** [comment]
I agree it should be a configuration option.

**Comment by eat-sleep-code** [comment]
Is there any update on providing an option for case-insensitive URLs?

**Comment by thomasgauvin** [comment]
Hi folks, as @miwebst indicated above, the case insensitivity is quite important and I think it's the generally accepted convention (https://www.w3.org/TR/WD-html40-970708/htmlweb.html), what is the use case for case insensitivity and what prevents you from having case sensitive files? Would love to better understand the use case regarding why this is needed

**Comment by tomaustin700** [comment]
Hi @thomasgauvin, in my case it was just a mistake however the fact that the deployment succeeded was more of an issue. If case sensitivity is required then I feel like the deployment should fail if any files don't match the required casing.

**Comment by thomasgauvin** [comment]
Thanks @tomaustin700, that's a good point

**Comment by eat-sleep-code** [comment]
> Hi folks, as @miwebst indicated above, the case insensitivity is quite important and I think it's the generally accepted convention (https://www.w3.org/TR/WD-html40-970708/htmlweb.html), what is the use case for case insensitivity and what prevents you from having case sensitive files? Would love to better understand the use case regarding why this is needed and why your architecture leads to URL references that are not case-sensitive

Very straight forward use case.   Much of the web has operated in a case-insensitive manner for URLs, regardless of what "spec" may say.   IIS was case-insensitive and powered large portions of the website.   Kestrel is also case-insensitive if deployed on a Windows host.

Many web sites _are_ case-insensitive.   For example, if I go to https://microsoft.com/AZURE it works the same as https://microsoft.com/azure 

Think about it from a human behavior.    If you have a URL that someone is going to type in, do you expect they will follow case sensitivity?   I mean how many people still type "www." before a domain if you tell them to go to something like "expedia.com" 

By _forcing_ case-sensitivity we are looking at only from a machine-focused mindset and not a human mindset.    

While case-sensitive should remain the default, there should be a config option to disable case-sensitivity.
